### PR TITLE
[flutter_tools] add skeleton for build uwp

### PR DESCRIPTION
**Pull request by jonahwilliams**
Adds the rest of the scaffolding for building a UWP application. The actual build functionality needs to be implemented, but could use buildWindows as an example (if it is going through cmake)

https://github.com/flutter/flutter/issues/14967

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/build_winuwp.dart [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  @override
+  final String name = 'winuwp';
+
+  @override
+  bool get hidden => !featureFlags.isWindowsUwpEnabled || !globals.platform.isWindows;
+
+  @override
+  Future<Set<DevelopmentArtifact>> get requiredArtifacts async => <DevelopmentArtifact>{
+    // TODO(flutter): add a windows_uwp artifact here once that is updated.
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/commands/build_winuwp.dart:34]: Is there an issue that describes what needs to be done here in some more detail?

File: packages/flutter_tools/lib/src/windows/build_windows.dart [hunk 2]
```diff
@@ -110,6 +113,28 @@ Future<void> buildWindows(WindowsProject windowsProject, BuildInfo buildInfo, {
...
+/// Note that this feature is currently unfinished.
+Future<void> buildWindowsUwp(WindowsUwpProject windowsProject, BuildInfo buildInfo, {
+  String target,
+  VisualStudio visualStudioOverride,
+}) async {
+  if (!windowsProject.existsSync()) {
+    throwToolExit(
+      'No Windows UWP desktop project configured. See '
+      'https://flutter.dev/desktop#add-desktop-support-to-an-existing-flutter-app '
+      'to learn about adding Windows support to a project.');
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/windows/build_windows.dart:127]:
```suggestion
      'to learn about adding Windows support to a project.',
    );
```

File: packages/flutter_tools/lib/src/windows/build_windows.dart [hunk 3]
```diff
@@ -110,6 +113,28 @@ Future<void> buildWindows(WindowsProject windowsProject, BuildInfo buildInfo, {
...
+  }
+  if (windowsProject.projectVersion != kCurrentUwpTemplateVersion) {
+    throwToolExit(
+      'The Windows UWP project template and build process has changed. In order to build '
+      'you must delete the winuwp directory and re-create the project.'
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/windows/build_windows.dart:132]:
```suggestion
      'you must delete the winuwp directory and re-create the project.',
```

File: packages/flutter_tools/lib/src/commands/build_winuwp.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/build_winuwp.dart:34]: Done

**Review by zanderso — APPROVED** [review]
tool code lgtm, but tbh I'm not especially clear on where we are in the process or what the next steps are.

**Comment by jonahwilliams** [comment]
@cbracken is working on the getting the engine bits, and I'm just trying to set things up for @clarkezone on the tool side of things.

**Comment by jonahwilliams** [comment]
g3 failure is a flake, landing

**Comment by clarkezone** [comment]
Awesome, thx @jonahwilliams

**Pull request merged**